less_users/views.py

```diff
@@ -75,69 +87,95 @@ def get_queryset(self):
     def get_context_data(self, **kwargs):
         """adds all challenges by logged in user and active ones to the context"""
         context = super().get_context_data(**kwargs)
-        context['my_all'] = self.get_queryset()
-        context['my_active'] = self.get_queryset().filter(is_active=True)
+        context["my_all"] = self.get_queryset()
+        context["my_active"] = self.get_queryset().filter(is_active=True)
         return context
 
     def get(self, request, **kwargs):
         """adds sorting options (ordering by...) of user_challenge items by: >active< >date< or >alphabetically<"""
         my_all = self.get_queryset()
         my_active = self.get_queryset().filter(is_active=True)
-        if request.GET.get('order_value'):
-            order_value = request.GET.get('order_value')
+        if request.GET.get("order_value"):
+            order_value = request.GET.get("order_value")
             my_all = my_all.order_by(order_value)
-        return render(request, 'less_users/my_challenges.html', context={'my_all': my_all, 'my_active': my_active})
+        return render(
+            request,
+            "less_users/my_challenges.html",
+            context={"my_all": my_all, "my_active": my_active},
+        )
 
     def post(self, request, **kwargs):
-        return redirect('my_challenges')
+        return redirect("my_challenges")
+
+
+class MyChallengeView(LoginRequiredMixin, DetailView):
+    """GET: show details of challenge user has taken;
+    POST: for active challenges- allows adding points to user's Profile (total points)
+    and generate Log object by connecting UserChallenge with now-date;
+    for all challenges- allows deleting"""
+
+    model = UserChallenge
+    template_name = "less_users/my_challenge.html"
 
 
 def activate_view(request, pk):
     """it checks if there exists active user_challenge for logged in user and chosen challenge;
     if not- it creates new active user_challenge"""
-    challenge_id = int(request.POST.get('activate'))
-    user_challenge_exist = \
-        UserChallenge.objects.filter(user=request.user, challenge_id=challenge_id, is_active=True)
-    if len(user_challenge_exist) == 0:
-        user_new_challenge = UserChallenge.objects.create(user=request.user, challenge_id=challenge_id)
-        user_new_challenge.save()
-        messages.success(request, 'You activated new challenge!')
-        return HttpResponseRedirect(reverse('challenge_detail', args=[str(pk)]))
+    user_challenge_exist = UserChallenge.objects.filter(
+        user=request.user, challenge_id=pk, is_active=True
+    )
+    if user_challenge_exist.exists():
```

Comment:
```suggestion
    user_challenge = UserChallenge.objects.filter(
        user=request.user, challenge_id=pk, is_active=True
    )
    if user_challenge.exists():
```

Reply:
poprawiona nazwa zmiennej (bez _exist, bo to dopiero sprawdzam niżej)